File: app/src/main/java/com/syedmuzani/umbrella/adapters/MainPageListAdapter.java
package com.syedmuzani.umbrella.adapters;

import android.app.Activity;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.TextView;

import com.syedmuzani.umbrella.R;
import com.syedmuzani.umbrella.models.MainMenuLink;

import java.util.List;


public class MainPageListAdapter extends BaseAdapter {

    private static final String TAG = "MainPageListAdapter";

    private Activity activity;
    private List<MainMenuLink> links;

    public MainPageListAdapter(Activity activity, List<MainMenuLink> links) {
        this.activity = activity;
        this.links = links;
    }

    @Override
    public int getCount() {
        return links.size();
    }

    @Override
    public Object getItem(int position) {
        return position;
    }

    @Override
    public long getItemId(int position) {
        return position;
    }

    @Override
    public View getView(final int position, View view, ViewGroup parent) {
        view = activity.getLayoutInflater().inflate(R.layout.list_item, null);
        MainMenuLink link = links.get(position);
        TextView tv = (TextView) view.findViewById(R.id.tv);
        tv.setText(link.getTitle());
        return view;
    }
}

File: app/src/main/java/com/syedmuzani/umbrella/adapters/MainPageListAdapter.kt
package com.syedmuzani.umbrella.adapters

import android.content.Intent
import android.support.v7.widget.RecyclerView
import android.support.v7.widget.RecyclerView.Adapter
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import com.syedmuzani.umbrella.models.MainMenuLink
import org.jetbrains.anko.find


class MainPageListAdapter(private val items: List<MainMenuLink>): Adapter<MainPageListAdapter.ViewHold>() {
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHold {
        val view = LayoutInflater.from(parent.context)
                .inflate(android.R.layout.simple_list_item_1, parent, false)
        return ViewHold(view)
    }

    override fun onBindViewHolder(holder: ViewHold, position: Int) {
        holder.bindItems(items[position])
    }

    override fun getItemCount(): Int = items.size

    class ViewHold(view: View) : RecyclerView.ViewHolder(view) {
        private val tv: TextView = view.find(android.R.id.text1)
        private val context = itemView.context

        fun bindItems(link: MainMenuLink) {
            tv.text = link.title
            tv.setOnClickListener {
                val intent = Intent(context, link.activityClass)
                context.startActivity(intent)
            }
        }
    }

    companion object {
        private const val TAG = "MainPageListAdapter"
    }
}